common/lib/xmodule/xmodule/js/src/video/09_video_caption.js

```diff
@@ -352,7 +357,11 @@ function (Sjson, AsyncProcess) {
                     // for availability other transcripts.
                     if (_.keys(state.config.transcriptLanguages).length > 1) {
                         self.fetchAvailableTranslations();
-                    } else {
+                    }
+                    else if (!fetch_with_youtubeId && state.videoType === 'html5'){
+                        self.fetchCaption(true);
```

Comment:
Add log statement here to tell that you are Re-trying to fetch captions with youtubeId.